### PR TITLE
A "Watch Node" script

### PR DESCRIPTION
**Pull request by cmc333333**
README describes what this does, but the hope is that it'd make debugging a little bit easier. Right now, it's possible to manually grep through trees and notices, but it's awfully cumbersome.

This has a lot of duplication with `build_from.py` and doesn't make use of checkpoints, but I think it's still useful in its current form.

Looks like:

```
$ python watch_node.py 1005-16-c fr-notices/articles/xml/201/131/725.xml 12
WARNING:root:Bad format for whole appendix
> 2011-31725
    Added: (c) Notice requirement. To meet the requirements of paragraph (b) of this section, an automated teller machine operator must comply with the following:
> 2013-06861
    Modified: (c) Notice requirement. An automated teller machine operator must provide the notice required by paragraph (b) of this section either by showing it on the screen of the automated teller machine or by providing it on paper, before the consumer is committed to paying a fee.
```

### CODE REVIEW

**Comment by cmc333333** [comment]
bump

File: regparser/builder.py [hunk 1]
```diff
@@ -74,6 +74,21 @@ def gen_and_write_layers(self, reg_tree, act_info, cache, notices=None):
                 layer)
 
     def revision_generator(self, reg_tree):
+        for version, merged_changes in self.changes_in_sequence():
```
Comment (khandelwal) [inline regparser/builder.py]: A docstring here, will greatly help explain what this function does .

File: regparser/notice/changes.py [hunk 2]
```diff
@@ -240,6 +241,31 @@ def remove_intro(l):
...
+            return verb + ' (title: %s): %s' % (node['title'], node['text'])
+        else:
+            return verb + ": " + node['text']
+    elif change['action'] == Verb.DELETE:
+        return 'Deleted'
+    elif change['action'] == Verb.DESIGNATE:
+        return 'Moved to ' + '-'.join(change['destination'])
+    elif change['action'] == Verb.RESERVE:
+        return 'Reserved'
+    elif change['action'] == 'KEEP':
```
Comment (khandelwal) [inline regparser/notice/changes.py]: KEEP doesn't seem to be a Verb (as in Verb.KEEP). Should it be? Even if it's a no-op?

File: watch_node.py [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+from regparser.builder import Builder
+from regparser.notice.changes import node_to_dict, pretty_change
+from regparser.tree.struct import find
+
+
+def init_reg_and_builder(args):
+    """This function repeats a lot of what's done in build_from.py -- parse
+    the provided regulation file into a tree and create a Builder object"""
+    reg_text = ''
+    with codecs.open(args.filename, 'r', 'utf-8') as f:
```
Comment (khandelwal) [inline watch_node.py]: Should this share code with build_from? So that, if we ever change how that works, this will change too?

**Comment by khandelwal** [comment]
Should this optionally have a verbose mode, where it also prints out the AMDPAR statement?

**Comment by khandelwal** [comment]
This is a great addition. I've left a few minor comments, but in general am happy to merge this.

File: regparser/builder.py (hunk 1 above)
Comment (cmc333333) [inline regparser/builder.py]: Roger

File: regparser/notice/changes.py (hunk 2 above)
Comment (cmc333333) [inline regparser/notice/changes.py]: Good idea; I'll check it out

File: watch_node.py (hunk 3 above)
Comment (cmc333333) [inline watch_node.py]: I'll play with it a bit to see what can be pulled out

**Comment by cmc333333** [comment]
> Should this optionally have a verbose mode, where it also prints out the AMDPAR statement?

Great idea. Let's delay it for a later release, though, as it requires a significantly deeper dive

**Comment by cmc333333** [comment]
Updated to combine some shared code, add comments, and create a `Verb.KEEP`

**Pull request merged**